### PR TITLE
Index file based distributedLoad

### PR DESCRIPTION
**Pull request by chaowangnk1**

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 1]
```diff
@@ -248,10 +264,11 @@ private void addJob(URIStatus status, int replication) {
    *
    * @param filePath The {@link AlluxioURI} path to load into Alluxio memory
    * @param replication The replication of file to load into Alluxio memory
+   * @param indexFile file The index file containing all files to load
    */
-  private void distributedLoad(AlluxioURI filePath, int replication)
+  private void distributedLoad(AlluxioURI filePath, int replication, String indexFile)
```
Comment (bastrik) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java:269]: Do you check the validity of the index file?

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 2]
```diff
@@ -64,6 +66,16 @@
...
 
   private static final int DEFAULT_ACTIVE_JOBS = 1000;
+  private static final Option INDEXFILE_OPTION =
+      Option.builder()
+          .longOpt("indexfile")
+          .required(false)
+          .hasArg(true)
+          .numberOfArgs(1)
+          .type(Number.class)
+          .argName("replicas")
```
Comment (bastrik) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java]: Should arg name be replicas here?

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java (hunk 1 above)
Comment (chaowangnk1) [reply shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java:269]: RandomAccessFile will throw clear exception in cases the file can't be read.

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java (hunk 2 above)
Comment (chaowangnk1) [reply shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java]: updated

**Review by bastrik — APPROVED** [review]

**Pull request merged**